.github/workflows/amd-offline-tests.yml

```diff
@@ -22,48 +43,91 @@ jobs:
-          apt install -y libpython3-dev python3-pip git wget
-          # get rocm related utilities
-          wget https://repo.radeon.com/amdgpu-install/5.5/ubuntu/jammy/amdgpu-install_5.5.50500-1_all.deb
-          apt-get install -y ./amdgpu-install_5.5.50500-1_all.deb
-          amdgpu-install -y --usecase=rocm --no-dkms
-          # remove unused packages to free some disk space
-          cp /opt/rocm/llvm/bin/ld.lld .
-          apt purge -y rocblas rocm-llvm
-          mkdir -p /opt/rocm/llvm/bin
-          mv ld.lld /opt/rocm/llvm/bin/ld.lld
-          # install pytorch
-          pip3 install torch==2.0.1 --index-url https://download.pytorch.org/whl/rocm5.4.2
-          # uninstall system triton to eliminate conflicts with testing version
-          pip3 uninstall -y pytorch-triton-rocm
-
-      - name: Install Triton
+          pip3 uninstall -y triton
+
+      - name: Set up Python Virtual Environment
+        run: |
+          python3 -m venv ./venv_triton_upstream
+          source ./venv_triton_upstream/bin/activate
+  
+      - name: Install Dependencies
         run: |
-          cd python
-          pip3 install ninja
-          # Install in system, because need to override system triton. Otherwise lit tests will use wrong version
-          DEBUG=TRUE TRITON_USE_ROCM=TRUE TRITON_USE_ASSERT_ENABLED_LLVM=TRUE python3 -m pip install --no-build-isolation -vvv -e .
+          source ./venv_triton_upstream/bin/activate
+          python -m pip install --upgrade pip
+          python -m pip install ninja wheel pytest pytest-xdist numpy lit autopep8 flake8 isort
+          python -m pip install scipy>=1.7.1
+          python -m pip install cmake==3.24
+          python -m pip install torch==1.13.1 --index-url https://download.pytorch.org/whl/rocm5.2 # important for detecting ROCM!
+      
+      - name: Check pre-commit
+        # TODO: enable pre-commit tests
+        if: false
+        run: |
+          python -m pip install --upgrade pre-commit
+          python -m pre_commit run --all-files --verbose
 
-      - name: Run lit tests
+      - name: Set branch name
         run: |
-          python3 -m pip install lit
-          cd python
-          LIT_TEST_DIR="build/$(ls build | grep -i cmake)/test"
-          if [ ! -d "${LIT_TEST_DIR}" ]; then
-            echo "Could not find '${LIT_TEST_DIR}'" ; exit -1
+          if [ "${{ github.event_name }}" = "pull_request" ]; then
+            echo "BRANCH_NAME=${{ github.head_ref }}" >> $GITHUB_ENV
+          else
+            echo "BRANCH_NAME=$(git rev-parse --abbrev-ref HEAD)" >> $GITHUB_ENV
           fi
-          lit -v "${LIT_TEST_DIR}"
 
-      - name: Run CXX unittests
+      - name: Build Upstream with Branch as Backend
         run: |
-          cd python
-          cd "build/$(ls build | grep -i cmake)"
-          ctest
+          source ./venv_triton_upstream/bin/activate
+          set -x
+          ROOT_DIR=`realpath "."`
+          SCRIPT_PATH=$ROOT_DIR/scripts/amd
```

Comment:
looks like this is not used and this dir is no longer there. Let;s remove for now